src/Microsoft.DotNet.Darc/src/DarcLib/AzureDevOpsClient.cs

```diff
@@ -884,6 +887,55 @@ public async void StartNewReleaseAsync(string accountName, string projectName, A
                 body,
                 versionOverride: "5.0-preview.3",
                 baseAddressSubpath: "vsrm.");
+
+            return content.GetValue("id").ToObject<int>();
+        }
+
+        /// <summary>
+        /// Block until the release with ID informed finishes execution.
+        /// Checks the status of the release every 15 seconds.
+        /// </summary>
+        /// <param name="accountName"></param>
+        /// <param name="projectName"></param>
+        /// <param name="releaseId"></param>
+        public void WaitUntilCompleted(string accountName, string projectName, int releaseId)
+        {
+            while (true)
```

Comment:
You need some level of try/catch error handling here so that intermittent issues (e.g. random 500 from AzDO) don't cause havoc.

I'd also put in a max timeout.

Reply:
Yeah, I was looking for exactly that kind of feedback here!